### ISSUE TITLE
Preserve non-lightweight Label

### ISSUE THREAD

**Issue by KarlSevenSeven** [twinbasic/twinbasic#1248]
**Is your feature request related to a problem? Please describe.**
Not a problem but a small wish.
Before 146 the label was non-lightweight.
Good it is now!

The good thing with the non-lightweight Label is it can be zordered.

**Describe the solution you'd like**
Preserve the non-lightweight Label.
Or an additional property for non-lightweight.

**Describe alternatives you've considered**
Using a textbox when zorder 0 is needed.
Or other form design.

**Additional context**
-

**Comment by Kr00l** [comment]
If @WaynePhillipsEA decides to give green light..then my 2 cents:

1. Don't give a new property to enable non-lightweight as a lightweight works quite different. (BackStyle, shared DC etc.) It would be confusing to have other properties work differently and behave differently. (Zorder etc)
2. Make a new 'WindowedLabel' control from scratch.

**Comment by WaynePhillipsEA** [comment]
As @Kr00l mentions, the implementation is quite different, so I agree the only reasonable option would be to offer two separate controls.

This seems a worthwhile idea, but I will await any other comments/objections before proceeding with this.

One concern is... with regards to the other lightweight controls... wouldn't we want the same there?

**Comment by mburns08109** [comment]
Is having a ZOrder'd label worth the overhead of an entire control?
What's the real benefit? (I'm just not seeing it)
Perhaps a form property to set the ZOrder for ALL the "lightweight controls" may be a better option to explore?

**Comment by Kr00l** [comment]
The old solution was to put a Label on a borderless Frame or PictureBox container to have a window to solve ZOrder.

**Comment by WaynePhillipsEA** [comment]
> Perhaps a form property to set the ZOrder for ALL the "lightweight controls" may be a better option to explore?

That's not possible.  Lightweight controls are rendered as part of the container, i.e. always beneath the normal windowed controls.

**Comment by bclothier** [comment]
> The old solution was to put a Label on a borderless Frame or PictureBox container to have a window to solve ZOrder.

Why not a disabled textbox styled as a label?

**Comment by Kr00l** [comment]
> > The old solution was to put a Label on a borderless Frame or PictureBox container to have a window to solve ZOrder.
> 
> Why not a disabled textbox styled as a label?

The TextBox can't emulate the Label 100%.
See DT_EDITCONTROL on DrawText API.

**Comment by sokinkeso** [comment]
In my opinion, it is better to have the basic set of controls for compatibility with vb6, and later can be created a second set with additional controls more enhanced (windowed-label, image-buttons etc).
As for the image control that is windowless , picture is doing its job as windowed control. If picturebox had a "stretch" property then everything is ok.

**Comment by DinyaZ** [comment]
I think VB6 compatibility was the main goal, so it is important. And if TWB above this main goal creates extra functions, new controls, it will be a bonus point.

**Comment by fafalone** [comment]
Would definitely like to see it stick around. It's useful and the code is already written (minus the small task of changing it's name/guids so it's separate). More options are always a good thing.

Not sure if we need a non-lightweight Image control though as it's basically a less capable PictureBox at that point. Are their advantages over a PictureBox I'm not thinking of?

**Comment by WaynePhillipsEA** [comment]
Yes, Image is probably covered by PictureBox.  So probably the only other one to consider is Shape.

**Comment by wqweto** [comment]
Btw, windowless controls are supposed to work in containers that does not support windowless activation (like MS Access) so theoretically `Image` and `Label` should handle being windowed.

Is it possible to have `Windowless` **public** property on these controls?

This would need some serious OLE Controls trickery but would allow labels being windowless (by default) on forms with hundreds of labels and still allow special ones to be z-ordered between other non-windowless controls on the form.

**Comment by sokinkeso** [comment]
If I'm not mistaken, the major benefit of windowless controls is the rendering speed, and the transparency they offer. So it depends on the application if a windowed or windowless control is better to use. Since Wayne has already designed a second set of controls (custom Wayne controls), if these can be later exist on a normal form, we will all be happy 😊

**Comment by rexxitall** [comment]
> Would definitely like to see it stick around. It's useful and the code is already written (minus the small task of changing it's name/guids so it's separate). More options are always a good thing.
> 
> Not sure if we need a non-lightweight Image control though as it's basically a less capable PictureBox at that point. Are their advantages over a PictureBox I'm not thinking of?

Based on my expierience currently i think definitly we need something better then stdpicture :/ stdpicture can not change it size, the handle - a almost useless thing in my opinion. But i am not the master of desaster in that area. A control which would had help me would be some kind of felxible image which can also just stuck in the memory. Can change its size, can be filled, and has some basic draw features (all kinds of Blitting) . No fancy scaling, just a transparent pixel bitmap without a frame. (Adding alpha channels is also a pain). Ok i might be to stupid or new to this. The image should have a device context if needed and the handle should be able to be read write. So basicly a stdpicture without hussle. Aside my work a lot of game developers will have a need for. Such a thing is ideal as sprite, playground whatever. And for the testing. The fun starts with VB6 if you has more then 2500 of them. I was thinking for speed reasons to create all of them one times with device handle and so son. Was not a great idea. VB6 crashes. Curent soloution is to create compatible memory bitmaps and add device context just as needed. I like how JS handle such kind of images. They can be clicked and hoovered if they are visible. Can get focus etc. Such a thing would be really a gift. ;)

**Comment by fafalone** [comment]
@rexxitall There's existing UserControls like that. Of course only 32bit VB6 ones you'd need to compile into ocx. For now. It will be a glorious day when we can add and use UserControls from .tbcontrol.
